docs/deployment/teleport.md

```diff
@@ -0,0 +1,24 @@
+# Authentication with Teleport
+
+Setting up Teleport authentication will protect all routes of your notebook. It is particularly useful for adding authentication to deployed notebooks. Teleport authentication is provided in addition to [Livebook's authentication](../authentication.md) for authoring notebooks.
+
+## How to
+
+To integrate Teleport authentication with Livebook,
+set the `LIVEBOOK_IDENTITY_PROVIDER` environment variable to `LIVEBOOK_IDENTITY_PROVIDER=teleport:https://teleport-cluster.example.com/.well-known/jwks.json`, make sure the `teleport` CLI is installed and available on your machine (or your Docker image).
```

Comment:
Also, for simplicity, I would make it so the user only has to configure `https://teleport-cluster.example.com/` and we will append "/.well-known/jwks.json" using URI.append_path.